3 reports fpmembers.sql

```diff
@@ -0,0 +1,48 @@
+use FitPlexDB
+go 
+
+/*
+1) I would like a list of all active members with their membership details. I would like each member to be listed as last name, first initial.
+*/
+
+select concat(MemberLastName, ', ', substring(MemberFirstName,1,1), '.') as MemberName, MemberPhoneNum, MemberInitialWeightLb, MemberGoalWeightLb, MemberCurrentWeightLb, MemberBMI,
+MembershipStartDate, MembershipEndDate, MembershipType, PaymentStatus
+from Member
+where MembershipEndDate > getdate()
+
+/*
+	2) I want a list of all active members whose memberships are due to expire within the next month so that I can text them a 	reminder to renew their subscription.
+*/
+
+select * 
+from Member 
+where MembershipEndDate between getdate() and DATEADD(month,1,getdate())
+
+/*
+	3) I want to collect payments that are past due, but would first like to see how much I am owed. 
+    Please provide a count of all members per membership type and show how many paid and how many did not. 
+
+*/
+
+select Members = count(*), MembershipType, PaymentStatus
+from Member
+group by MembershipType, PaymentStatus
+
+
+/*
+	4) I would like to check in on the overall effectiveness of our gym's weight loss program. 
+	Please provide a list of all members and their progress, sorted by the amount of weight loss in desc order. 
+*/
+
+select MemberFirstName, MemberLastName, MemberWeightLostLb
+from member
+order by MemberWeightLostLb DESC
+
+/*
+	5) I would like to see the difference in progress made by members in different membership categories. 
+	Please show me the average weight loss per membership type.
+*/
+
```

Comment:
Since you only put in each member once, this works. However if you enter each member several times for each session they came, you would need to sum up their weight lost to get a accurate picture.

Reply:
Not sure what you mean... the column in the equation, MemberWeightLost, is a calculated column that calculates the weight lost. I would appreciate it if you could explain what the problem would be as I don't really get it.

Reply:
Sorry this comment is on the wrong report. It is meant to be on the 4th report, where you are supposed to show the progress of each member. The way you are doing it, it would only show the progress for each session separately, it wouldn't show the full progress of the member in one row.

Reply:
Thank you for clarifying which report your comment refers to. However, I'm still not sure what needs to be changed. It seems to me that what the report is asking for is the current amount of weight lost by each member of the gym, sorted from most to least. I think that displaying the ''MemberWeightLostLb" column would accomplish that, as it automatically takes the data from the ''CurrentWeight'' column, which is a record of the most recent weight, and subtracts the "InitialWeight", giving you an accurate picture of the progress made by that member. 
Is the "CurrentWeight" column the problem, as it only allows for data to be recorded once, rather than at each session?

Reply:
The weight difference is not recorded from session to session; it is calculated in total from "InitialWeight" to "CurrentWeight"

Reply:
No the current weight is fine, but if you want to know the total that the member lost, you would have to sum up all his MemberWeightLostLb rows from all session into one row.

Reply:
No the current weight is fine, but if you want to know the total that the member lost, you would have to sum up all his MemberWeightLostLb rows from all session into one row. We want to see what the members progress is through all his sessions. So the report should group be member and sum up the MemberWeightLostLb. Maybe if you add a few session for one  member in the data, it will be easier to follow what I mean.

Reply:
Thank you for the explanation, but I'm still not getting it... The MemberWeightLostLb column does not record weight loss progress from one session to another. It records progress from the InitialWeight, which is the client's weight from before they started attending the gym, until the CurrentWeight, and updates automatically every time a new CurrentWeight is entered, providing an accurate picture of the total amount of weight lost. Summing up the data in the MemberWeightLostLb column each time a new session is entered would provide a huge number that is completeley inaccurate.

Reply:
Your right! The way I thought you are using the InitialWeight was to record the weight at the begging of the session. I guess if you use it the way you are saying that it records the same weight every time then your query is correct.